picosat.c

```diff
@@ -4714,7 +4713,7 @@ collect_clauses (PS * ps)
 	      Ltk * lstk = LIT2IMPLS (lit);
 	      Lit ** r, ** s;
 	      r = lstk->start;
-	      if (lstk->count && (lit->val != TRUE || LIT2VAR (lit)->level))
+	      if (r && (lit->val != TRUE || LIT2VAR (lit)->level))
```

Comment:
`r` (and later `s`) are aliases for `lstk->start` or offsets into its owned storage. If lstk->start is NULL at 4716, then `r` will be at the if check, so the condition will fail and the contained for loop be skipped.

Since lstk->count is clearly a "how many elements" metric on lstk->start, it is either 0 or garbage if lstk->start is NULL.

Reply:
...if it's garbage, the original could would SEGV, yes? As noted above, it seems that `!start` ⇒ `!count` and this change has no functional effect. If that impression is incorrect, I'd really like to understand *how*.

(Your correct-AFAICT observation that `count` "is either 0 or garbage" would seem to confirm the stated implication.)

Reply:
Reverted my changes in favor of text from PR#1.